File: src/main/java/com/libraryapp/domain/user/LoanBookHistory.java
package com.libraryapp.domain.user;

import jakarta.persistence.*;

@Entity
public class LoanBookHistory {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private Users user;

    private String bookName;

    private boolean isReturn;

    public LoanBookHistory() {}

    public LoanBookHistory(Users user, String bookName, boolean isReturn) {
        this.user = user;
        this.bookName = bookName;
        this.isReturn = isReturn;
    }

    public void doReturn() {
        this.isReturn = true;
    }

    public Long getId() {
        return id;
    }

    public Users getUser() {
        return user;
    }

    public String getBookName() {
        return bookName;
    }

    public boolean isReturn() {
        return isReturn;
    }
}

File: src/main/kotlin/com/libraryapp/domain/user/loanhistory/LoanBookHistory.kt
package com.libraryapp.domain.user.loanhistory

import com.libraryapp.domain.user.Users
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class LoanBookHistory(
    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: Users,

    val bookName: String,

    @Enumerated(EnumType.STRING)
    var status: UserLoanStatus = UserLoanStatus.LOANED,

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
) {
    fun doReturn() {
        this.status = UserLoanStatus.RETURNED
    }

    companion object {
        fun fixture(
            user: Users,
            bookName: String = "kotlin",
            status: UserLoanStatus = UserLoanStatus.LOANED,
            id: Long? = null
        ): LoanBookHistory {
            return LoanBookHistory(
                user = user,
                bookName = bookName,
                status = status,
                id = id
            )
        }
    }
}